src/debugger/ios/iOSPlatform.ts

```diff
@@ -57,10 +63,10 @@ export class IOSPlatform implements IAppPlatform {
                     Q(IOSPlatform.RUN_IOS_FAILURE_PATTERNS)).process(runIosSpawn);
         }
 
-        return new Compiler(this.projectPath).compile().then(() => {
-            return new DeviceDeployer(this.projectPath).deploy();
+        return new Compiler(this.iosProjectPath).compile().then(() => {

```

Comment:
I think you probably need to modify Compiler, DeviceDeployer, IOSDebugModeManager, and DeviceRunner so they don't append the "ios" subfolder any more to the paths...